docs/architecture/adr-003-subaccounts.md

```diff
@@ -0,0 +1,88 @@
+# ADR 3: Module SubAccounts
+
+## Changelog
+
+## Context
+
+Currently `ModuleAccount`s must be declared upon supply keeper initialization. In addition to this they don't allow for separation of fungible coins within an account.
+
+New structs should be added so a `ModuleAccount` can dynamically add accounts.
```

Comment:
Instead of stating new structs should be added here and under `## Decision`, we should rather state that the desire to support the ability to define and manage sub-module-accounts.

Reply:
updated, let me know if there is anything else you think should be added